File: Backend/Data/Models/Project.cs
using Data.Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Data.Models
{
    public class Project : BaseEntity
    {
        [Required]
        [MaxLength(200)]
        public string Name { get; set; } = string.Empty;

        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public ProjectStatus Status { get; set; } = ProjectStatus.Active;

        [Column("CompanyId")]
        public new int CompanyId { get; set; }
        public Company Company { get; set; } = null!;

        // Navigation properties
        public ICollection<Issue> Issues { get; set; } = new List<Issue>();
    }
}

File: Frontend/timeTrackerFlutter/lib/core/models/project.dart
import 'package:json_annotation/json_annotation.dart';
import '../enums/project_status.dart';

part 'project.g.dart';

@JsonSerializable()
class Project {
  const Project({
    required this.id,
    required this.companyId,
    required this.companyName,
    required this.name,
    required this.status,
    required this.issueCount,
    required this.createdAt,
    this.startDate,
    this.endDate,
  });

  factory Project.fromJson(Map<String, dynamic> json) =>
      _$ProjectFromJson(json);

  final int id;
  final int companyId;
  final String companyName;
  final String name;
  final DateTime? startDate;
  final DateTime? endDate;
  final int status;
  final int issueCount;
  final DateTime createdAt;

  ProjectStatus get projectStatus => ProjectStatus.fromValue(status);

  Map<String, dynamic> toJson() => _$ProjectToJson(this);
}
